pyiceberg/table/metadata.py

```diff
@@ -292,6 +292,13 @@ def snapshot_by_name(self, name: str) -> Optional[Snapshot]:
             return self.snapshot_by_id(ref.snapshot_id)
         return None
 
+    def _snapshot_as_of_timestamp_ms(self, timestamp_ms: int) -> Optional[Snapshot]:
```

Comment:
Any reason not to make this public?

Reply:
Getting a snapshot by timestamp should be a public function, I'm not opposed to making this public. But I'm unsure if `timestamp_ms: int` is the preferred signature we want as input.

Reply:
timestamp_ms certainly keeps us closer to the spec, although I could see a case for `timestamp_ms: int | datetime.datetime` (maybe requiring datetimes to be timezone-aware?).

Reply:
I'm +1 for just supporting `timestamp_ms: int` for now, because it is what we support consistently across all Iceberg APIs, and as a bonus we don't have to worry about validating timezone-awareness.

Reply:
Thanks! Looks like it was already implemented in #748